### PR TITLE
fix: transaction loop wait when http connection unavailable

### PR DESCRIPTION
**Pull request by sulenn**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/api_handler.go  1
         

Clones removed
==============
+ client/api_handler.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6037399&bid=19783850)

File: client/api_handler.go [hunk 1]
```diff
@@ -111,16 +111,19 @@ func (api *APIHandler) SendRawTransaction(ctx context.Context, groupID int, tx *
...
-		logger := log.New("hash", tx.Hash())
 		for {
 			receipt, err := api.GetTransactionReceipt(ctx, groupID, tx.Hash().Hex())
 			if receipt != nil {
 				return receipt, nil
 			}
 			if err != nil {
-				logger.Trace("Receipt retrieval failed", "err", err)
+				errorStr := fmt.Sprintf("%s", err)
+				if strings.Contains(errorStr, "connection refused") { // judge key word "connection refused"
```
Comment (bxq2011hust) [inline client/api_handler.go:121]:
```suggestion
				if strings.Contains(errorStr, "connection refused") {
```

**Review by bxq2011hust — DISMISSED** [review]

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/api_handler.go  1
         

Clones removed
==============
+ client/api_handler.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6037399&bid=19783850)

**Review by bxq2011hust — DISMISSED** [review]

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/api_handler.go  1
         

Clones removed
==============
+ client/api_handler.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6037399&bid=19783850)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- client/api_handler.go  1
         

Clones removed
==============
+ client/api_handler.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6037399&bid=19783850)

**Review by bxq2011hust — APPROVED** [review]

**Pull request merged**